test/benchmarks/src/main/scala/scala/collection/mutable/OpenHashMapBenchmark.scala

```diff
@@ -0,0 +1,204 @@
+    private[this] val nanosPerPut: Double = 5
+
+    /** Minimum number of nanoseconds per invocation, so as to avoid timing artifacts. */
+    private[this] val minNanosPerInvocation = 1000000  // one millisecond
+
+    /** Size of the maps created in this trial. */
+    private[this] var size: Int = _
+
+    /** Total number of entries in all of the `maps` combined. */
+    var mapEntries: Int = _
+
+    /** Number of operations performed in the current invocation. */
+    var operations: Int = _
+
+    /** Bytes of memory used in the object graphs of all the maps. */
+    var memory: Long = _
+
+    var maps: Array[OpenHashMap[Int,Int]] = null
+
+    @Setup
+    def threadSetup(params: BenchmarkParams) {
+      size = params.getParam("size").toInt
+      val n = math.ceil(minNanosPerInvocation / (nanosPerPut * size)).toInt
+      mapEntries = size * n
+      maps = new Array(n)
+    }
+
+    @Setup(Level.Iteration)
+    def iterationSetup {
+      operations = 0
+    }
+
+    @Setup(Level.Invocation)
+    def setup(params: IterationParams) {
+      for (i <- 0 until maps.length) maps(i) = new OpenHashMap[Int,Int](size)
+
+      if (params.getType == IterationType.MEASUREMENT) {
+        operations += mapEntries
+        System.gc()  // clean up after last invocation
+      }
+    }
+
+    @TearDown(Level.Iteration)
+    def iterationTeardown(params: IterationParams) {
+      if (params.getType == IterationType.MEASUREMENT) {
+        // limit to smaller cases to avoid OOM
+        memory = if (mapEntries <= 1000000) GraphLayout.parseInstance(maps(0), maps.tail).totalSize else 0
+      }
+    }
+  }
+
+  /** State container for the `get()` bulk calling tests.
+    * 
+    * Provides a thread-scoped map of the expected size.
+    * Performs a GC after loading the map.
+    */
+  @State(Scope.Thread)
+  class BulkGetState {
+    val map = new OpenHashMap[Int,Int].empty
+
+    /** Load the map with keys from `1` to `size`. */
+    @Setup
+    def setup(params: BenchmarkParams) {
+      val size = params.getParam("size").toInt
+      put_Int(map, 1, size)
+      System.gc()
+    }
+  }
+
+  /** State container for the `get()` bulk calling tests with deleted entries.
+    * 
+    * Provides a thread-scoped map of the expected size, from which entries have been removed.
+    * Performs a GC after loading the map.
+    */
+  @State(Scope.Thread)
+  class BulkRemovedGetState {
+    val map = new OpenHashMap[Int,Int].empty
+
+    /** Load the map with keys from `1` to `size`, removing half of them. */
+    @Setup
+    def setup(params: BenchmarkParams) {
+      val size = params.getParam("size").toInt
+      put_remove_Int(map, size)
+      System.gc()
+    }
+  }
+
+  /** Put elements into the given map. */
+  private def put_Int(map: OpenHashMap[Int,Int], from: Int, to: Int) {
+    var i = from
+    while (i <= to) {  // using a `for` expression instead adds significant overhead
+      map.put(i, i)
+      i += 1
+    }
+  }
```

Comment:
You end here without doing anything with your constructed map.  This is generally safe enough, but it's probably good practice to include a dump of the result to Blackhole (e.g. the size) to be sure that the work isn't optimized away.  (Especially if someone copies and pastes this code for another benchmark where the putting is changed into a more elidable operation.)